server/api/api/endpoints/functions.py

```diff
@@ -751,3 +751,67 @@ async def _get_function_status(data, auth_info: mlrun.common.schemas.AuthInfo):
             HTTPStatus.BAD_REQUEST.value,
             reason=f"Runtime error: {err_to_str(err)}",
         )
+
+
+@router.get("/projects/{project}/functions/by_foo_spec/{foo_spec}")
+async def list_functions_by_foo_spec(
```

Comment:
This function is duplicated from `list_functions`. Create a common function that can be called from both places.